### PR TITLE
Add Classify in SoftmaxRegression

### PR DESCRIPTION
**Pull request by micyril**
Fix for [#935](https://github.com/mlpack/mlpack/issues/935)

### CODE REVIEW

File: src/mlpack/methods/softmax_regression/softmax_regression.hpp [hunk 1]
```diff
@@ -113,12 +113,25 @@ class SoftmaxRegression
    * calculates the probabilities for every class, given a data point. It then
    * chooses the class which has the highest probability among all.
    *
+   * This method is deprecated -- you should use Classify() instead.
```
Comment (rcurtin) [inline src/mlpack/methods/softmax_regression/softmax_regression.hpp]: We can mark that we'll remove this in mlpack 3.0.0.

File: src/mlpack/methods/softmax_regression/softmax_regression_impl.hpp [hunk 2]
```diff
@@ -121,14 +121,22 @@ void SoftmaxRegression<OptimizerType>::Predict(const arma::mat& testData,
 }
 
 template<template<typename> class OptimizerType>
+void SoftmaxRegression<OptimizerType>::Classify(const arma::mat& dataset,
+                                                arma::Row<size_t>& labels)
+    const
+{
+  Predict(dataset, labels);
+}
```
Comment (rcurtin) [inline src/mlpack/methods/softmax_regression/softmax_regression_impl.hpp]: We can change `Predict()` to wrap `Classify()` instead; that will make the transition easier to remove `Predict()` for 3.0.0.

**Comment by micyril** [comment]
Ryan,

I hadn't found any call of `Predict` in `softmax_regression_test.cpp`, `ComputeAccuracy` was used there instead. But you are right, I forgot about softmax_regression_main.cpp, I will add changes shortly.

**Comment by rcurtin** [comment]
Great, thanks.  We can mark the method `MLPACK_DEPRECATED` to issue a compiler warning when a user uses the function; I did this for softmax regression in d109399.  I realized that logistic regression isn't marked `MLPACK_DEPRECATED` so I also did that in cf06a76.

I noticed that your name isn't in `src/mlpack/core.hpp` and `COPYRIGHT.txt`, would you like to add it to the list of contributors?  Also it seems the email you are committing with isn't linked to your Github account, maybe you want to fix that?  I don't know if it is something that matters to you. :)

**Comment by micyril** [comment]
Thank you for suggestions. I have linked the email to my GitHub account and have sent a [PR](https://github.com/mlpack/mlpack/pull/949) for adding my name to `src/mlpack/core.hpp` and `COPYRIGHT.txt`.

**Pull request merged**